Plugins/Triage/Collect-WindowsArtifacts.ps1

```diff
@@ -0,0 +1,318 @@
+
+    } elseif (-not $7z64bitTestPath) 
+    {
+        Throw "7za_x64.exe not detected in Bin. Place executable in Bin directory and try again."
+    }
+
+    foreach ($Computer in $ComputerName) 
+    {
+        #Handle $ComputerName defined as localhost and not break the Plugin
+        if ($Computer -eq "Localhost") 
+        {
+            $Computer = $ENV:ComputerName
+        }
+
+        #Verify machine is online and ready for data collection
+        if (-not (Test-Connection -ComputerName $Computer -Count 1 -Quiet)) 
+        {
+            Write-Error ("{0} appears to be offline. Cannot collect artifacts." -f $Computer)
+            Continue
+        }
+
+        $7zaInstallexe = $7za64
+
+        #Establish PS Session criteria
+        $Session = New-PSSession -ComputerName "$Computer" -SessionOption (New-PSSessionOption -NoMachineProfile)
+         
+        #Create Output directory structure on remote host
+        $RemoteDumpPath = ("\\{0}\c`$\ProgramData\{0}" -f $Computer)
+        If (-not (Test-Path $RemoteDumpPath)) 
+        {
+            New-Item -Type Directory -Path $RemoteDumpPath | Out-Null
+        }
+
+        #Copy velociraptor to remote host
+        $SmbPathVelociraptor = ("\\{0}\c`$\ProgramData\{1}" -f $Computer, (Split-Path -Path $Velociraptor -Leaf))
+        
+        try 
+        {
+            Copy-Item -Path $Velociraptor -Destination $SmbPathVelociraptor -ErrorAction Stop
+            $RemoteFileVelociraptor = Get-Item -Path $SmbPathVelociraptor -ErrorAction Stop
+
+            # verify that the file copy succeeded to the remote host
+            if (-not $RemoteFileVelociraptor) 
+            {
+                Write-Error ("Velociraptor not found on {0}. There may have been a problem during the copy process. Artifacts were not acquired." -f $Computer)
+                Continue
+            }
+
+        } 
+        catch 
+        {
+            Write-Error ("An unexpected error occurred while copying velociraptor to {0}. Artifacts not acquired." -f $Computer)
+            Continue
+        }
+
+        #Copy 7za.exe to remote system
+        $SmbPath7za = ("\\{0}\c`$\ProgramData\{1}" -f $Computer, (Split-Path -Path $7zaInstallexe -Leaf))
+        
+        try 
+        {
+            Copy-Item -Path $7zaInstallexe -Destination $SmbPath7za -ErrorAction Stop
+            $RemoteFile7za = Get-Item -Path $SmbPath7za -ErrorAction Stop
+
+            #verify that the file copy succeeded to the remote host
+            if (-not $RemoteFile7za) 
+            {
+                Write-Error ("7za_x64.exe not found on {0}. There may have been a problem during the copy process. Artifacts cannot be compressed." -f $Computer)
+                Continue
+            }
+        } 
+        catch 
+        {
+            Write-Error ("An unexpected error occurred while copying 7za_x64.exe to {0}. Artifacts cannot be compressed." -f $Computer)
+            Continue
+        }
+
```

Comment:
This section of script block is very functional and specific on what each command is doing. I feel like there is a way to bundle these commands together in a `$ScriptBlock` so you only have to make one `Invoke-Command` call, but after fumbling around with it for the last hour or so I couldn't figure it out....So anyway, this looks great!